### PR TITLE
feat: endpoints for seedlot status

### PR DESCRIPTION
**Pull request by annibalsilva**
# Description

Create endpoints to fetch seed lot status, which will be all represented in the system by enums for now.

Fixes [FSADT2-491](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-491)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- Manually
- Unit tests

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

### CODE REVIEW

**Comment by annibalsilva** [comment]
@mgaseta, can you check the [endpoints documentation](https://nr-spar-oracle-api-79-service-api.apps.silver.devops.gov.bc.ca/swagger-ui/index.html#/seedlot)? I'd also like your opinion on whatever fetching by code will be useful or not (see the [Jira task](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-491) for a short summary).

File: src/main/java/ca/bc/gov/backendstartapi/endpoint/SeedlotStatusEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import ca.bc.gov.backendstartapi.enumeration.SeedlotStatus;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.util.MimeTypeUtils;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/** Endpoints to fetch {@link SeedlotStatus}. */
+@RestController
+@RequestMapping(path = "/api/seedlot-status", produces = MimeTypeUtils.APPLICATION_JSON_VALUE)
+@Tag(name = "seedlot")
```
Comment (RMCampos) [inline src/main/java/ca/bc/gov/backendstartapi/endpoint/SeedlotStatusEndpoint.java:12]: I think the tag should point to the endpoint, in this case `SeedlotStatus`.

File: src/main/java/ca/bc/gov/backendstartapi/config/SwaggerConfig.java [hunk 2]
```diff
@@ -51,6 +54,27 @@ public OpenAPI theRestApi() {
                         .type(Type.HTTP)
                         .scheme("bearer")
                         .bearerFormat("JWT")))
-        .addSecurityItem(new SecurityRequirement().addList(BEARER_SECURITY_SCHEME_NAME));
+        .addSecurityItem(new SecurityRequirement().addList(BEARER_SECURITY_SCHEME_NAME))
+        // TODO: find a way to keep this closer to the controller itself.
+        .paths(
+            new Paths()
+                .addPathItem(
+                    "/api/seedlot-status",
```
Comment (RMCampos) [inline src/main/java/ca/bc/gov/backendstartapi/config/SwaggerConfig.java:62]: Paths are discovered automatically by spring-doc. Don't need to register them in the code.

**Review by RMCampos — APPROVED** [review]
Great work! Small changes and we're good to go!

**Review by RMCampos — CHANGES_REQUESTED** [review]
Oops. Clicked the wrong button. Once those changes are addressed it's all good.

**Comment by RMCampos** [comment]
I was thinking here. Is that necessairy to "generify" them? Because we still are going to need an especific class for each enumeration. Also one class for each one. So I think we could go with the traditional style. Create a simple enum class containing code a description, and an endpoint class to hold the GET method and the path.

**Review by mgaseta — APPROVED** [review]
@annibalsilva The documentation looks great, and regarding the code fetch, it doesn't seem necessary at the moment, we also have an enum on the client that can map status, so it may be redundant to have this endpoint also. But it may have functionality for the project in the future that I can't imagine right now, so I don't think it should be removed yet.

**Comment by annibalsilva** [comment]
Reopened in [PR 53 of bcgov/nr-spar-backend](https://github.com/bcgov/nr-spar-backend/pull/53).

**Pull request closed without merging**